### PR TITLE
ci: push testbench images to SaaS registry

### PR DESCRIPTION
**Pull request by lenaschoenburg**
In SaaS we can't use images from gcr.io/zeebe/ anymore so we'll push them to the blessed SaaS registry instead.

The starter/worker images can remain on gcr.io because they are running outside of SaaS.

### CODE REVIEW

**Comment by lenaschoenburg** [comment]
AFAICT we have no way of testing this without merging to main. Since QA is broken anyway, there's no risk in merging this without testing before.

File: .github/workflows/zeebe-testbench.yaml [hunk 1]
```diff
@@ -62,7 +75,7 @@ jobs:
       - uses: ./.github/actions/build-platform-docker
         id: build-zeebe-docker
         with:
-          repository: gcr.io/zeebe-io/zeebe
+          repository: europe-docker.pkg.dev/camunda-saas-registry/zeebe/zeebe
```
Comment (than-pet) [inline .github/workflows/zeebe-testbench.yaml:78]:
```suggestion
          repository: europe-docker.pkg.dev/camunda-saas-registry/zeebe-io/zeebe
```

**Review by than-pet — CHANGES_REQUESTED** [review]

File: .github/workflows/zeebe-testbench.yaml (hunk 1 above)
Comment (lenaschoenburg) [reply .github/workflows/zeebe-testbench.yaml:78]: Are you sure? I can access europe-docker.pkg.dev/camunda-saas-registry/zeebe/zeebe but not europe-docker.pkg.dev/camunda-saas-registry/zeebe-io/zeebe

Comment (than-pet) [reply .github/workflows/zeebe-testbench.yaml:78]: Yes, you can try again.
It will work now.

**Review by than-pet — APPROVED** [review]

**Pull request merged**